### PR TITLE
Fix ordering in course dropdown

### PR DESCRIPTION
**Pull request by balderdash**
2 changes:

* Minecraft adventurer and designer weren't being ordered correctly because they have hardcoded friendly names. Figure out dropdown positions based on the actual script/course name instead.
* Move the minecraft tutorials and applab-intro to the top of the hoc category

### CODE REVIEW

File: lib/cdo/script_constants.rb [hunk 1]
```diff
@@ -186,7 +186,7 @@ def self.assignable_info(course_or_script)
       # Would better be called something like assignable_name
       script_name: course_or_script[:name],
       category: first_category,
-      position: ScriptConstants.position_in_category(name, first_category),
+      position: ScriptConstants.position_in_category(course_or_script[:name], first_category),
```
Comment (joshlory) [inline lib/cdo/script_constants.rb:189]: Why this change?

Comment (balderdash) [reply lib/cdo/script_constants.rb:189]: For minecraft adventurer and designer, `name` is "Minecraft Adventurer" and "Minecraft Designer", whereas the script names are `mc` and `minecraft`. The script names are used in the category hash that specifies all the ordering, so until now they've been incorrectly appearing at the bottom of the hoc category.

I added  a unit test for that particular bug.
